### ISSUE TITLE
Rule Request: file_name_no_space prevent whitespace characters in file names

### ISSUE THREAD

**Issue by timcmiller** [realm/SwiftLint#3007]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Prevent any whitespace characters in file names. Xcode allows you to name files with spaces, but it causes issues (or annoyances) when trying to get a path to those files since you have to escape the space.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

While there is a lot of disagreement on weather file names should be camelCase(Xcode default), kabab-case, or snake_case. There is a general consensus that there should not be spaces in file names. [Apple](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingBasics.html), [Google](https://developers.google.com/style/filenames), [Smithsonian](https://library.si.edu/sites/default/files/tutorial/pdf/filenamingorganizing20180227.pdf)

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

Does not trigger
```
File.swift
File+Extension.swift
NSFile.swift
File_Name.swift
File-Name.swift
```

Does trigger
```
File Name.swift
```

3. Should the rule be configurable, if so what parameters should be configurable?

You would want to be able to exclude specific files, and I could see a use exclude certain suffixes as well. For example there may be files named `File+Test Extension.swift`.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Because Apple doesn't specifically call out no spaces in file names, it should be opt-in.
